nifi-marklogic-processors/src/main/java/org/apache/nifi/marklogic/processor/QueryMarkLogic.java

```diff
@@ -216,13 +215,11 @@ public void onTrigger(ProcessContext context, ProcessSessionFactory sessionFacto
             // Save a reference to this solely to facilitate unit testing
             this.queryBatcher = queryBatcherContext.getBatcher();
 
-            // Can transfer the incoming FlowFile immediately
-            session.transfer(incomingFlowFile, ORIGINAL);
-
-            runQueryBatcherAndCommit(session, new Tuple<DataMovementManager, QueryBatcher>(queryBatcherContext.getManager(), queryBatcherContext.getBatcher()));
+            runQueryBatcherJob(session, queryBatcherContext);
+            transferAndCommit(session, incomingFlowFile, ORIGINAL);
```

Comment:
This was a key change - gotta transfer the incomingFlowFile at the very end, once we know we don't need to create any new FlowFiles from it.